linera-views/src/graphql.rs

```diff
@@ -0,0 +1,853 @@
+    }
+
+    fn parse(value: Option<async_graphql::Value>) -> async_graphql::InputValueResult<Self> {
+        let Some(async_graphql::Value::Object(obj)) = value else {
+            return Err(async_graphql::InputValueError::expected_type(
+                value.unwrap_or_default(),
+            ));
+        };
+        Ok(Self {
+            filters: async_graphql::InputType::parse(obj.get("filters").cloned())
+                .map_err(async_graphql::InputValueError::propagate)?,
+        })
+    }
+
+    fn to_value(&self) -> async_graphql::Value {
+        let mut map = async_graphql::indexmap::IndexMap::new();
+        map.insert(async_graphql::Name::new("filters"), self.filters.to_value());
+        async_graphql::Value::Object(map)
+    }
+
+    fn federation_fields() -> Option<String> {
+        Some(format!(
+            // for each field
+            "{{ {0} }}",
+            if let Some(fields) = MapFilters::<K>::federation_fields() {
+                format!("{0} {1}", "filters", fields)
+            } else {
+                "filters".to_string()
+            }
+        ))
+    }
+
+    fn as_raw_value(&self) -> Option<&Self::RawValueType> {
+        Some(self)
+    }
+}
+
+use crate::map_view::ByteMapView;
+impl<C: Send + Sync, V: async_graphql::OutputType> async_graphql::TypeName for ByteMapView<C, V> {
+    fn type_name() -> Cow<'static, str> {
+        format!("ByteMapView_{}", V::type_name()).into()
+    }
+}
+#[async_graphql::Object(name_type)]
+impl<C, V> ByteMapView<C, V>
+where
+    C: Context + Send + Sync,
+    V: async_graphql::OutputType
+        + serde::ser::Serialize
+        + serde::de::DeserializeOwned
+        + Clone
+        + Send
+        + Sync
+        + 'static,
+    ViewError: From<C::Error>,
+{
+    #[graphql(derived(name = "keys"))]
+    async fn keys_(&self, count: Option<usize>) -> Result<Vec<Vec<u8>>, async_graphql::Error> {
+        let keys = self.keys().await?;
+        let it = keys.iter().cloned();
+        Ok(if let Some(count) = count {
+            it.take(count).collect()
+        } else {
+            it.collect()
+        })
+    }
+
+    async fn entry(&self, key: Vec<u8>) -> Result<Entry<Vec<u8>, Option<V>>, async_graphql::Error> {
+        Ok(Entry {
+            value: self.get(&key).await?,
+            key,
+        })
+    }
+
+    async fn entries(
+        &self,
+        input: Option<MapInput<Vec<u8>>>,
+    ) -> Result<Vec<Entry<Vec<u8>, Option<V>>>, async_graphql::Error> {
+        let keys = input
+            .and_then(|input| input.filters)
+            .and_then(|filters| filters.keys);
+        let keys = if let Some(keys) = keys {
+            keys
+        } else {
+            self.keys().await?
+        };
+
+        let mut values = vec![];
+        for key in keys {
+            values.push(Entry {
+                value: self.get(&key).await?,
+                key,
+            })
+        }
+
+        Ok(values)
+    }
+}
+
+use crate::map_view::MapView;
```

Comment:
I'm planning to move them into their respective modules after I do the API reduction — but for now chose to leave it all in one file because the boilerplatey nature of it means it often wants to be modified together.